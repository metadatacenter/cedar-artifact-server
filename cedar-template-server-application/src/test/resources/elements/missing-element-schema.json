{
  "@id": "https://repo.staging.metadatacenter.net/template-elements/e7a3f3f6-9802-4e22-8988-1a91ba230cdd",
  "@type": "https://schema.metadatacenter.org/core/TemplateElement",
  "@context": {
    "xsd": "http://www.w3.org/2001/XMLSchema#",
    "pav": "http://purl.org/pav/",
    "oslc": "http://open-services.net/ns/core#",
    "schema": "http://schema.org/",
    "pav:createdOn": {
      "@type": "xsd:dateTime"
    },
    "pav:createdBy": {
      "@type": "@id"
    },
    "pav:lastUpdatedOn": {
      "@type": "xsd:dateTime"
    },
    "oslc:modifiedBy": {
      "@type": "@id"
    }
  },
  "type": "object",
  "title": "Address element schema element schema",
  "description": "Address element schema element schema autogenerated by the CEDAR Template Editor 1.0.1-SNAPSHOT",
  "_ui": {
    "title": "Address element schema",
    "description": "Address element schema autogenerated by the CEDAR Template Editor",
    "order": [
      "streetFieldSchema",
      "cityFieldSchema",
      "zipFieldSchema",
      "countryFieldSchema"
    ]
  },
  "properties": {
    "@context": {
      "type": "object",
      "properties": {
        "pav": {
          "type": "string",
          "format": "uri",
          "enum": [
            "http://purl.org/pav/"
          ]
        },
        "oslc": {
          "type": "string",
          "format": "uri",
          "enum": [
            "http://open-services.net/ns/core#"
          ]
        },
        "streetFieldSchema": {
          "enum": [
            "https://schema.staging.metadatacenter.net/properties/streetFieldSchema"
          ]
        },
        "cityFieldSchema": {
          "enum": [
            "https://schema.staging.metadatacenter.net/properties/cityFieldSchema"
          ]
        },
        "zipFieldSchema": {
          "enum": [
            "https://schema.staging.metadatacenter.net/properties/zipFieldSchema"
          ]
        },
        "countryFieldSchema": {
          "enum": [
            "https://schema.staging.metadatacenter.net/properties/countryFieldSchema"
          ]
        }
      },
      "patternProperties": {
        "^(?!pav)(?!schema)(?!oslc)[a-zA-Z][a-zA-Z0-9]*$": {
          "type": "string",
          "format": "uri"
        }
      },
      "required": [
        "streetFieldSchema",
        "cityFieldSchema",
        "zipFieldSchema",
        "countryFieldSchema"
      ],
      "additionalProperties": false
    },
    "@id": {
      "type": [
        "string",
        "null"
      ],
      "format": "uri"
    },
    "@type": {
      "oneOf": [
        {
          "type": "string",
          "format": "uri"
        },
        {
          "type": "array",
          "minItems": 1,
          "items": {
            "type": "string",
            "format": "uri"
          },
          "uniqueItems": true
        }
      ]
    },
    "pav:createdOn": {
      "type": [
        "string",
        "null"
      ],
      "format": "date-time"
    },
    "pav:createdBy": {
      "type": [
        "string",
        "null"
      ],
      "format": "uri"
    },
    "pav:lastUpdatedOn": {
      "type": [
        "string",
        "null"
      ],
      "format": "date-time"
    },
    "oslc:modifiedBy": {
      "type": [
        "string",
        "null"
      ],
      "format": "uri"
    },
    "streetFieldSchema": {
      "@type": "https://schema.metadatacenter.org/core/TemplateField",
      "@context": {
        "xsd": "http://www.w3.org/2001/XMLSchema#",
        "pav": "http://purl.org/pav/",
        "oslc": "http://open-services.net/ns/core#",
        "schema": "http://schema.org/",
        "pav:createdOn": {
          "@type": "xsd:dateTime"
        },
        "pav:createdBy": {
          "@type": "@id"
        },
        "pav:lastUpdatedOn": {
          "@type": "xsd:dateTime"
        },
        "oslc:modifiedBy": {
          "@type": "@id"
        }
      },
      "type": "object",
      "title": "Street field schema field schema",
      "description": "Street field schema field schema autogenerated by the CEDAR Template Editor ",
      "_ui": {
        "title": "Street field schema",
        "description": "Street field schema autogenerated by the CEDAR Template Editor",
        "inputType": "textfield"
      },
      "_valueConstraints": {
        "requiredValue": false
      },
      "properties": {
        "@type": {
          "oneOf": [
            {
              "type": "string",
              "format": "uri"
            },
            {
              "type": "array",
              "minItems": 1,
              "items": {
                "type": "string",
                "format": "uri"
              },
              "uniqueItems": true
            },
            {
              "enum": [
                "@id"
              ]
            }
          ]
        },
        "@value": {
          "type": [
            "string",
            "null"
          ]
        },
        "_valueLabel": {
          "type": [
            "string",
            "null"
          ]
        },
        "pav:createdOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "pav:createdBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        },
        "pav:lastUpdatedOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "oslc:modifiedBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        }
      },
      "required": [
        "@value"
      ],
      "pav:createdOn": "2017-03-24T15:01:06-0700",
      "pav:createdBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "pav:lastUpdatedOn": "2017-03-24T15:01:06-0700",
      "oslc:modifiedBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "additionalProperties": false,
      "@id": "https://repo.staging.metadatacenter.net/template-fields/71afa763-210d-4402-a9ee-be0c9213ab83",
      "$schema": "http://json-schema.org/draft-04/schema#"
    },
    "cityFieldSchema": {
      "@type": "https://schema.metadatacenter.org/core/TemplateField",
      "@context": {
        "xsd": "http://www.w3.org/2001/XMLSchema#",
        "pav": "http://purl.org/pav/",
        "oslc": "http://open-services.net/ns/core#",
        "schema": "http://schema.org/",
        "pav:createdOn": {
          "@type": "xsd:dateTime"
        },
        "pav:createdBy": {
          "@type": "@id"
        },
        "pav:lastUpdatedOn": {
          "@type": "xsd:dateTime"
        },
        "oslc:modifiedBy": {
          "@type": "@id"
        }
      },
      "type": "object",
      "title": "City field schema field schema",
      "description": "City field schema field schema autogenerated by the CEDAR Template Editor ",
      "_ui": {
        "title": "City field schema",
        "description": "City field schema autogenerated by the CEDAR Template Editor",
        "inputType": "textfield"
      },
      "_valueConstraints": {
        "requiredValue": false
      },
      "properties": {
        "@type": {
          "oneOf": [
            {
              "type": "string",
              "format": "uri"
            },
            {
              "type": "array",
              "minItems": 1,
              "items": {
                "type": "string",
                "format": "uri"
              },
              "uniqueItems": true
            },
            {
              "enum": [
                "@id"
              ]
            }
          ]
        },
        "@value": {
          "type": [
            "string",
            "null"
          ]
        },
        "_valueLabel": {
          "type": [
            "string",
            "null"
          ]
        },
        "pav:createdOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "pav:createdBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        },
        "pav:lastUpdatedOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "oslc:modifiedBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        }
      },
      "required": [
        "@value"
      ],
      "pav:createdOn": "2017-03-24T15:01:06-0700",
      "pav:createdBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "pav:lastUpdatedOn": "2017-03-24T15:01:06-0700",
      "oslc:modifiedBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "additionalProperties": false,
      "@id": "https://repo.staging.metadatacenter.net/template-fields/b6d877dd-cada-4a96-bf84-f38906cacd54",
      "$schema": "http://json-schema.org/draft-04/schema#"
    },
    "zipFieldSchema": {
      "@type": "https://schema.metadatacenter.org/core/TemplateField",
      "@context": {
        "xsd": "http://www.w3.org/2001/XMLSchema#",
        "pav": "http://purl.org/pav/",
        "oslc": "http://open-services.net/ns/core#",
        "schema": "http://schema.org/",
        "pav:createdOn": {
          "@type": "xsd:dateTime"
        },
        "pav:createdBy": {
          "@type": "@id"
        },
        "pav:lastUpdatedOn": {
          "@type": "xsd:dateTime"
        },
        "oslc:modifiedBy": {
          "@type": "@id"
        }
      },
      "type": "object",
      "title": "Zip field schema field schema",
      "description": "Zip field schema field schema autogenerated by the CEDAR Template Editor ",
      "_ui": {
        "title": "Zip field schema",
        "description": "Zip field schema autogenerated by the CEDAR Template Editor",
        "inputType": "textfield"
      },
      "_valueConstraints": {
        "requiredValue": false
      },
      "properties": {
        "@type": {
          "oneOf": [
            {
              "type": "string",
              "format": "uri"
            },
            {
              "type": "array",
              "minItems": 1,
              "items": {
                "type": "string",
                "format": "uri"
              },
              "uniqueItems": true
            },
            {
              "enum": [
                "@id"
              ]
            }
          ]
        },
        "@value": {
          "type": [
            "string",
            "null"
          ]
        },
        "_valueLabel": {
          "type": [
            "string",
            "null"
          ]
        },
        "pav:createdOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "pav:createdBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        },
        "pav:lastUpdatedOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "oslc:modifiedBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        }
      },
      "required": [
        "@value"
      ],
      "pav:createdOn": "2017-03-24T15:01:06-0700",
      "pav:createdBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "pav:lastUpdatedOn": "2017-03-24T15:01:06-0700",
      "oslc:modifiedBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "additionalProperties": false,
      "@id": "https://repo.staging.metadatacenter.net/template-fields/74a68f13-da41-4c82-96ef-41aea6ad67b6",
      "$schema": "http://json-schema.org/draft-04/schema#"
    },
    "countryFieldSchema": {
      "@type": "https://schema.metadatacenter.org/core/TemplateField",
      "@context": {
        "xsd": "http://www.w3.org/2001/XMLSchema#",
        "pav": "http://purl.org/pav/",
        "oslc": "http://open-services.net/ns/core#",
        "schema": "http://schema.org/",
        "pav:createdOn": {
          "@type": "xsd:dateTime"
        },
        "pav:createdBy": {
          "@type": "@id"
        },
        "pav:lastUpdatedOn": {
          "@type": "xsd:dateTime"
        },
        "oslc:modifiedBy": {
          "@type": "@id"
        }
      },
      "type": "object",
      "title": "Country field schema field schema",
      "description": "Country field schema field schema autogenerated by the CEDAR Template Editor ",
      "_ui": {
        "title": "Country field schema",
        "description": "Country field schema autogenerated by the CEDAR Template Editor",
        "inputType": "textfield"
      },
      "_valueConstraints": {
        "requiredValue": false
      },
      "properties": {
        "@type": {
          "oneOf": [
            {
              "type": "string",
              "format": "uri"
            },
            {
              "type": "array",
              "minItems": 1,
              "items": {
                "type": "string",
                "format": "uri"
              },
              "uniqueItems": true
            },
            {
              "enum": [
                "@id"
              ]
            }
          ]
        },
        "@value": {
          "type": [
            "string",
            "null"
          ]
        },
        "_valueLabel": {
          "type": [
            "string",
            "null"
          ]
        },
        "pav:createdOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "pav:createdBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        },
        "pav:lastUpdatedOn": {
          "type": [
            "string",
            "null"
          ],
          "format": "date-time"
        },
        "oslc:modifiedBy": {
          "type": [
            "string",
            "null"
          ],
          "format": "uri"
        }
      },
      "required": [
        "@value"
      ],
      "pav:createdOn": "2017-03-24T15:01:06-0700",
      "pav:createdBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "pav:lastUpdatedOn": "2017-03-24T15:01:06-0700",
      "oslc:modifiedBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
      "additionalProperties": false,
      "@id": "https://repo.staging.metadatacenter.net/template-fields/7fc971a1-53dd-4020-950b-e70fd4997924",
      "$schema": "http://json-schema.org/draft-04/schema#"
    }
  },
  "required": [],
  "pav:createdOn": "2017-03-24T15:01:06-0700",
  "pav:createdBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
  "pav:lastUpdatedOn": "2017-03-24T15:01:06-0700",
  "oslc:modifiedBy": "https://metadatacenter.org/users/74c2d307-af38-43a3-9ca2-5c644d2a4e58",
  "additionalProperties": false
}