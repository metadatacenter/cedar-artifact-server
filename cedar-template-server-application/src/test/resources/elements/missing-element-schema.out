{"validates":"false","warnings":[],"errors":["object has missing required properties ([\"$schema\"])","Error validating artifact at path / against CEDAR sub-schema coreJSONSchemaFields.json"]}